src/AggregateRoot.php

```diff
@@ -67,6 +84,16 @@ public function getRecordedEvents(): array
         return $this->recordedEvents;
     }
 
+    private function state(): array
+    {
+        $class = new ReflectionClass($this);
+
+        return collect($class->getProperties(ReflectionProperty::IS_PUBLIC))
```

Comment:
Does this mean that if you want to use snapshots, all properties you want to use should be public?